app/src/main/java/fr/free/nrw/commons/category/CategoryClient.kt

```diff
@@ -92,8 +92,8 @@ class CategoryClient @Inject constructor(private val categoryInterface: Category
                 it.query()?.pages() ?: emptyList()
             }
             .map {
-                it.map { page -> page.title().replace(CATEGORY_PREFIX, "") }
+                it.filter { page -> !page.categoryInfo().isHidden
+                }.map { page -> page.title().replace(CATEGORY_PREFIX, "") }
```

Comment:
I think we need to be careful here as the `categoryInfo` object wouldn't be present in all the responses that are handled by `responseMapper`. So, we should do this:

```suggestion
                it.filter {
                    page -> page.categoryInfo() == null || !page.categoryInfo().isHidden
                }.map { page -> page.title().replace(CATEGORY_PREFIX, "") }
```

Not handling this is likely a reason for the tests failing in Travis CI.